server/db/interface.go

```diff
@@ -148,20 +149,117 @@ type MatchData struct {
 	Quantity  uint64
 	Rate      uint64
 	Status    order.MatchStatus
-	Sigs      order.Signatures
-	// cachedHash order.MatchID // TODO
+}
+
+// SwapData contains the data generated by the clients during swap negotiation.
+type SwapData struct {
+	SigMatchAckMaker []byte
+	SigMatchAckTaker []byte
+	ContractA        []byte
+	ContractACoinID  []byte
+	ContractATime    int64
+	ContractAAckSig  []byte // B's signature of contract A data
+	ContractB        []byte
+	ContractBCoinID  []byte
+	ContractBTime    int64
+	ContractBAckSig  []byte // A's signature of contract B data
+	RedeemACoinID    []byte
+	RedeemATime      int64
+	RedeemAAckSig    []byte // B's signature of redeem A data
+	RedeemBCoinID    []byte
+	RedeemBTime      int64
+	RedeemBAckSig    []byte // A's signature of redeem B data
+}
```

Comment:
I'm actually not yet sure what archiving would mean (moving to a different table or purging the data entirely?), but I agree it will be necessary.  In terms of this pg implementation, nulling the BYTEA data would work, otherwise the swap data could be in its own table (referenced by the matches table) and the rows could just be removed.

The only reason we would want to state in the spec the minimum time to keep old swap data is if there are API calls that retrieve or access this data.  Did you have other considerations in mind regarding the need to put archival policy in the spec?

Reply:
Once we purge the data, we can't prove that we upheld our end of the deal. This means that the archive limit is the longest that someone can claim foul play and we can prove otherwise.

Reply:
That's sort of what I was considering with the API calls (for showing proof), but your statement describes the core motivation I think.

Another thing to note regarding space savings is that the contracts can be pulled from the blockchain using the coinID.  So at some point in the swap, perhaps on `MatchComplete` or even earlier, the contract fields could be nulled because the swap would not have completed if the contracts were not matched with the contract coinID pkScripts.  In fact, possibly the only reasons to keep them in storage are for swap resumption and to have a separate record until redeem + swap confs when there is no longer a change for change reorganization and potentially changed contracts depending on coin malleability?

Reply:
Um, no they can't be pulled from the contracts since they have hashes of the contract, but they would be in the sig script for the redeem.